### PR TITLE
Move interactions into a sub-app

### PR DESCRIPTION
**Pull request by teneightfive**
Continuing with the folder structure work this moves all of the
interactions work into a sub folder.

## Issues

Addresses #74 and #78.

### CODE REVIEW

File: src/apps/interactions/controllers/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const interactions = require('./controllers/interactions.controller')
```
Comment (tyom) [inline src/apps/interactions/controllers/index.js]: Shouldn't this be `require('./interactions.controller')`? I noticed it in other feature apps.

File: src/apps/interactions/controllers/index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+const edit = require('./controllers/edit.controller')
+
+module.exports = {
+  interactions,
```
Comment (tyom) [inline src/apps/interactions/controllers/index.js]: Wonder if this should be called `add` and the details and common should go in its own controller file.

File: src/apps/interactions/router.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const router = require('express').Router()
+
+const detailsController = require('./controllers/details.controller')
+const editController = require('./controllers/edit.controller')
+
+router.get('/interaction/add-step-1/', detailsController.getAddStep1)
+router.post('/interaction/add-step-1/', detailsController.postAddStep1)
```
Comment (tyom) [inline src/apps/interactions/router.js]: These (and the ones below) could be grouped under `router.route` method.

File: src/apps/interactions/router.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+router.get('/interaction/:interactionId/*', detailsController.getCommon)
```
Comment (tyom) [inline src/apps/interactions/router.js]: Maybe use `router.param` to fetch interaction details by id.

File: src/apps/interactions/router.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+router.get('/interaction/:interactionId/details', detailsController.getInteractionDetails)
+
+router.get(['/interaction/add/', '/interaction/:interactionId/edit', '/interaction/edit/'], editController.editDetails)
+router.post(['/interaction/add/', '/interaction/:interactionId/edit', '/interaction/edit/'], editController.postDetails)
+
+module.exports = {
+  // path: '/contact',
```
Comment (tyom) [inline src/apps/interactions/router.js]: Should it be mounted under `/interactions`?

File: src/apps/interactions/router.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const router = require('express').Router()
+
+const detailsController = require('./controllers/details.controller')
+const editController = require('./controllers/edit.controller')
+
+router.param('interactionId', detailsController.getCommon)
+
+router
```
Comment (ztolley) [inline src/apps/interactions/router.js]: personal preference, but I like 

router.route('/company/edit/ltd/:id')
  .get(foreignController.editCommon, ltdController.editDetails)
  .post(foreignController.postDetails)

I like having the methods indented.

File: src/apps/interactions/router.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  .route('/add-step-1/')
+  .get(detailsController.getAddStep1)
+  .post(detailsController.postAddStep1)
+
+router
+  .route([
```
Comment (ztolley) [inline src/apps/interactions/router.js]: bit ugly

Comment (teneightfive) [inline src/apps/interactions/router.js]: Why? That's also not a very useful statement 🙄

File: src/apps/interactions/router.js (hunk 6 above)
Comment (teneightfive) [inline src/apps/interactions/router.js]: We haven't used this pattern elsewhere. I'm sticking with the current pattern.

If we decide to switch we can do it across the board.

**Comment by teneightfive** [comment]
There are a few things within the interactions app that could be refactored/tidied up but I don't want to bundle it with this change and let the PR snowball.

We may also feel that interactions is best suited within company in the same style we have for investment and contacts.

**Pull request merged**